cmd/kube-proxy/app/server_test.go

```diff
@@ -36,36 +36,9 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	componentbaseconfig "k8s.io/component-base/config"
-	"k8s.io/component-base/configz"
 	kubeproxyconfig "k8s.io/kubernetes/pkg/proxy/apis/config"
 )
 
-// This test verifies that NewProxyServer does not crash when CleanupAndExit is true.
```

Comment:
This test fails because with the current approach we try to connect to the apiserver.
It is work to refactor to mock the clientset?

Sidenote, I was trying to understand when this options is being used but I could not find anything

Reply:
You can run `kube-proxy --cleanup` to clean up any existing iptables/ipvs rules and then immediately exit.

It looks like that test was a regression test for #14736 and probably isn't super necessary. If we're going to test it at all we should test that it actually removes iptables rules correctly...

Reply:
it is not great solution but as you said

> probably isn't super necessary.